Cleaned up the sftp parsing logic.

ckemper67 opened the pull request:
Simplified and cleaned up the sftp parsing logic. Added support to path.Clean the directory. Added additional tests.

ckemper67 committed:
Cleaned up the sftp parsing logic.

Simplified and cleaned up the sftp parsing logic. Added support to
path.Clean the directory. Added additional tests.

fd0 commented on src/restic/backend/sftp/config.go:
```diff
@@ -11,58 +12,47 @@ type Config struct {
+// ParseConfig parses the string s and extracts the sftp config. The two
+// supported configuration formats are sftp:user@host:port/directory and
+// sftp:user@host:directory. The directory will be path Cleaned and can be
+// an absolute path.
```
I think the comment is not accurate: What about the format `sftp://user@host`? The prefix isn't mentioned at all...

fd0 commented:
Hey, thanks for the contribution! Looks really good already, just a small nitpick: Could you correct the comment?

ckemper67 commented:
I corrected the comment and added a sentence about how to pass an absolute
directory since that is not completely obvious.

On Sun, Feb 21, 2016 at 10:20 AM, Alexander Neumann <
notifications@github.com> wrote:

> Hey, thanks for the contribution! Looks really good already, just a small
> nitpick: Could you correct the comment?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/restic/restic/pull/466#issuecomment-186879212>.
>
